src/Mod/Assembly/JointObject.py

```diff
@@ -946,6 +949,24 @@ def getIcon(self):
 
         return ":/icons/Assembly_CreateJoint.svg"
 
+    def getOverlayIcons(self):
+        """
+        Return a dictionary of overlay icons.
+        Keys are positions from Gui.IconPosition.
+        Values are the icon resource names.
+        """
+
+        overlays = {}
+
+        assembly = self.app_obj.Proxy.getAssembly(self.app_obj)
+        # Assuming Reference1 corresponds to the first part link
+        if hasattr(self.app_obj, "Reference1"):
+            part = UtilsAssembly.getMovingPart(assembly, self.app_obj.Reference1)
+            if part is not None and not assembly.isPartConnected(part):
+                overlays[Gui.IconPosition.BottomLeft] = "Part_Detached"
```

Comment:
Non-blocking comment, of course: 

Why not just `return { Gui.IconPosition.BottomLeft: "Part_Detached" }` and `return {}` at the end?

Reply:
Well I thought if we want to add other overlay laters it would be better like this.